Release KDM for 2.5

kinarashah opened the pull request.

Commit:
Add liveness and readiness probes to coredns autoscaler

Commit:
Add liveness and readiness probes to coredns autoscaler

Commit:
Go generate

Commit:
Go generate

Commit:
Merge pull request #821 from annablender/coredns-autoscaler-live-check-2.5

[Backport 2.5] Coredns autoscaler live check 2.5

Commit:
Fix coredns autoscaler probe config

Commit:
Fix coredns autoscaler probe config

Commit:
Go generate

Commit:
Go generate

Commit:
Merge pull request #832 from annablender/coredns-autoscaler-CI-fix-2.5

[Backport 2.5] Coredns autoscaler probe config fix

brandond committed:
Fix output buffering for Drone and improve error text

Signed-off-by: Brad Davidson <brad.davidson@rancher.com>
(cherry picked from commit d3b1d7d705a52bc3d9de4fc03e6ea65bc10496e2)

brandond committed:
Fix output buffering for Drone and improve error text

Signed-off-by: Brad Davidson <brad.davidson@rancher.com>
(cherry picked from commit d3b1d7d705a52bc3d9de4fc03e6ea65bc10496e2)

kinarashah committed:
remove extra space in ingress constraint

kinarashah committed:
Merge pull request #839 from kinarashah/ing25

briandowns committed:
February rke2 1.20 R2  (#834)

briandowns committed:
February rke2 1.20 R2  (#834)

doflamingo721 committed:
Copy k8s template

doflamingo721 committed:
Copy k8s template

doflamingo721 committed:
Add k8s template

doflamingo721 committed:
Add k8s template

doflamingo721 committed:
go generate

doflamingo721 committed:
go generate

doflamingo721 committed:
Merge pull request #844 from doflamingo721/2.5-feb-patch

[2.5] Add February k8s patches

kinarashah committed:
Revert "Add k8s template"

This reverts commit 6058d2999bb68fc7a291f72cc82771c431885501.

kinarashah committed:
Revert "Add k8s template"

This reverts commit 6058d2999bb68fc7a291f72cc82771c431885501.

kinarashah committed:
Revert "Copy k8s template"

This reverts commit 0343f6c9c1e5796750882ce45099df079a5f9a56.

kinarashah committed:
Revert "Copy k8s template"

This reverts commit 0343f6c9c1e5796750882ce45099df079a5f9a56.

kinarashah committed:
go generate

kinarashah committed:
go generate

kinarashah committed:
Merge pull request #849 from kinarashah/remove120

kinarashah committed:
base copy k8s 1.19.16, 1.20.15 and nginx template

kinarashah committed:
base copy k8s 1.19.16, 1.20.15 and nginx template

kinarashah committed:
add k8s 1.19.16 and 1.20.15 for new nginx template

kinarashah committed:
add k8s 1.19.16 and 1.20.15 for new nginx template

kinarashah committed:
go generate

kinarashah committed:
go generate

kinarashah committed:
Merge pull request #851 from kinarashah/cfg25

kinarashah committed:
update default k8s version

kinarashah committed:
update default k8s version

kinarashah committed:
go generate

kinarashah committed:
go generate

kinarashah committed:
Merge pull request #866 from kinarashah/k8s25

kinarashah committed:
Merge pull request #867 from kinarashah/dev2

rancher-max reviewed (dismissed):
rke2 update looks good and there isn't anything additional here for k3s, which is correct.
LGTM

kinarashah commented on rke/k8s_rke_system_images.go:
```diff
@@ -6174,6 +6218,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
+		"v1.20.15-rancher1-2": {
+			Etcd:                      "rancher/mirrored-coreos-etcd:v3.4.15-rancher1",
+			Kubernetes:                "rancher/hyperkube:v1.20.15-rancher1",
+			Alpine:                    "rancher/rke-tools:v0.1.79",
```
rke-tools in 2.6 has been bumped to v0.1.80, either this needs to be updated here or 2.6 needs a copy of this k8s version

kinarashah commented on rke/k8s_rke_system_images.go:
```diff
@@ -6174,6 +6218,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
+			AciControllerContainer:    "noiro/aci-containers-controller:5.1.1.0.1ae238a",
+			AciGbpServerContainer:     "noiro/gbp-server:5.1.1.0.1ae238a",
+			AciOpflexServerContainer:  "noiro/opflex-server:5.1.1.0.1ae238a",
+			PodInfraContainer:         "rancher/mirrored-pause:3.2",
```
same for mirrored-pause

kinarashah committed:
update images for 1.20.15-rancher1-2

system images should be consistent with dev-v2.6
so clusters don't autoupgrade on upgrade to 2.6 from 2.5

kinarashah committed:
go generate

kinarashah committed:
Merge pull request #873 from kinarashah/dev25t
